test/controllers/testOnly/ReportOneLoginProblemControllerSpec.scala

```diff
@@ -0,0 +1,319 @@
+      verifyNoInteractions(hmrcDeskproConnector)
+
+      val document = Jsoup.parse(contentAsString(result))
+      document.getElementsByClass("govuk-error-summary").size() should be > 0
+      document
+        .body()
+        .select("form[id=error-feedback-form]")
+        .first
+        .attr("action")                                       shouldBe s"/test-only/report-one-login-problem"
+    }
+
+    "return Bad Request and page with validation error if the name has invalid characters" in new TestScope {
+      val request = generateRequest(
+        isAjaxRequest = false,
+        name = """<a href="blah.com">something</a>"""
+      )
+
+      val submit = controller.submit(None)(request)
+      val page   = Jsoup.parse(contentAsString(submit))
+
+      status(submit) shouldBe BAD_REQUEST
+      verifyNoInteractions(hmrcDeskproConnector)
+
+      page.getElementsByClass("govuk-error-summary").size() should be > 0
+    }
+
+    "return Bad Request and page with validation error if the email has invalid syntax (for Deskpro)" in new TestScope {
+      val request = generateRequest(isAjaxRequest = false, email = "a.a.a")
+      val submit  = controller.submit(None)(request)
+      val page    = Jsoup.parse(contentAsString(submit))
+
+      status(submit) shouldBe BAD_REQUEST
+      verifyNoInteractions(hmrcDeskproConnector)
+
+      page.getElementsByClass("govuk-error-summary").size() should be > 0
+    }
+
+    "return Internal Server Error and error page if the Deskpro ticket creation fails" in new TestScope {
+      when(
+        hmrcDeskproConnector.createDeskProTicket(
+          meq("John Densmore"),
+          meq("name@mail.com"),
+          meq(controller.problemMessage("Some Action", "Some Error")),
+          meq("/contact/problem_reports"),
+          meq(false),
+          any[Request[AnyRef]](),
+          meq(None),
+          meq(Some("one-login-complaint")),
+          meq(None),
+          any[TicketConstants]
+        )
+      ).thenReturn(Future.failed(new Exception("failed")))
+
+      val request = generateRequest(isAjaxRequest = false)
+      val result  = controller.submit(None)(request)
+      status(result) should be(INTERNAL_SERVER_ERROR)
+
+      val document = Jsoup.parse(contentAsString(result))
+      document.text() should include("Try again later.")
+    }
+  }
+
+  "Requesting the standalone thanks page" should {
+    "return OK and valid html" in new TestScope {
+      val result = controller.thanks()(FakeRequest())
+
+      status(result) should be(OK)
+
+      val document = Jsoup.parse(contentAsString(result))
+      document.getElementsByClass("govuk-error-summary").size() should be(0)
+      document.getElementsByClass("govuk-panel__title").text()  should be(
+        "We have received your technical problem report"
+      )
+    }
+  }
+
+  class TestScope extends MockitoSugar {
+
+    val enrolmentsConnector: EnrolmentsConnector = mock[EnrolmentsConnector]
```

Comment:
This can be deleted as it's not used by this controller.

Reply:
Removed